core/src/main/groovy/com/novoda/gradle/release/AndroidArtifacts.groovy

```diff
@@ -1,41 +1,48 @@
 package com.novoda.gradle.release
 
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.tasks.bundling.Jar
 import org.gradle.api.tasks.javadoc.Javadoc
 
-class AndroidArtifacts implements Artifacts {
+class AndroidArtifacts {
 
-    def all(String publicationName, Project project) {
-        [sourcesJar(publicationName, project), javadocJar(publicationName, project), mainJar(project)]
+    private Project project
+
+    AndroidArtifacts(Project project) {
+        this.project = project
+    }
+
+    def all(String publicationName) {
+        [sourcesJar(publicationName), javadocJar(publicationName), mainJar()]
     }
 
-    def sourcesJar(String publicationName, Project project) {
+    def sourcesJar(String publicationName) {
         project.task(publicationName + 'AndroidSourcesJar', type: Jar) {
             classifier = 'sources'
             from project.android.sourceSets.main.java.srcDirs
         }
     }
 
-    def javadocJar(String publicationName, Project project) {
-        def androidJavadocs = project.task(publicationName + 'AndroidJavadocs', type: Javadoc) {
+    def javadocJar(String publicationName) {
+        Task androidJavadocs = project.task(publicationName + 'AndroidJavadocs', type: Javadoc) {
             source = project.android.sourceSets.main.java.srcDirs
-            classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
+            classpath += project.android.bootClasspath
             classpath += project.android.libraryVariants.toList().first().javaCompile.classpath
             classpath += project.android.libraryVariants.toList().first().javaCompile.outputs.files
         }
 
         project.task(publicationName + 'AndroidJavadocsJar', type: Jar, dependsOn: androidJavadocs) {
             classifier = 'javadoc'
-            from androidJavadocs.destinationDir
+            from androidJavadocs
```

Comment:
We only depend on the outputs of the `androidJavadocs` now, which currently happens to be `destinationDir`.